azurerm/resource_arm_operational_insight_workspace_test.go

```diff
@@ -0,0 +1,167 @@
+					testCheckAzureRMOperationalInsightWorkspaceExists("azurerm_operational_insight_workspace.test"),
+				),
+			},
+		},
+	})
+}
+func TestAccAzureRMOperationalInsightWorkspace_optional(t *testing.T) {
+	ri := acctest.RandInt()
+	config := testAccAzureRMOperationalInsightWorkspace_optional(ri, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMOperationalInsightWorkspaceDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMOperationalInsightWorkspaceExists("azurerm_operational_insight_workspace.test"),
+				),
+			},
+		},
+	})
+}
+
+func testCheckAzureRMOperationalInsightWorkspaceDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*ArmClient).workspacesClient
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "azurerm_operational_insight_workspace" {
+			continue
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup := rs.Primary.Attributes["resource_group_name"]
+
+		resp, err := conn.Get(resourceGroup, name)
+
+		if err != nil {
+			return nil
+		}
+
+		if resp.StatusCode != http.StatusNotFound {
+			return fmt.Errorf("OperationalInsight Workspace still exists:\n%#v", resp)
+		}
+	}
+
+	return nil
+}
+
+func testCheckAzureRMOperationalInsightWorkspaceExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		// Ensure we have enough information in state to look up in API
+		rs, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+
+		name := rs.Primary.Attributes["name"]
+		resourceGroup, hasResourceGroup := rs.Primary.Attributes["resource_group_name"]
+		if !hasResourceGroup {
+			return fmt.Errorf("Bad: no resource group found in state for OperationalInsight Workspace: '%s'", name)
+		}
+
+		conn := testAccProvider.Meta().(*ArmClient).workspacesClient
+
+		resp, err := conn.Get(resourceGroup, name)
+		if err != nil {
+			return fmt.Errorf("Bad: Get on OperationalInsight Workspace Client: %+v", err)
+		}
+
+		if resp.StatusCode == http.StatusNotFound {
+			return fmt.Errorf("Bad: OperationalInsight Workspace '%s' (resource group: '%s') does not exist", name, resourceGroup)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMOperationalInsightWorkspace_requiredOnly(rInt int, location string) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_operational_insight_workspace" "test" {
+  name                = "acctest-%d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku                 = "Free"
+}
+`, rInt, location, rInt)
+}
+
+func testAccAzureRMOperationalInsightWorkspace_optional(rInt int, location string) string {
```

Comment:
given this sets all of the fields we'd tend to call this `_Complete` rather than Optional - as such could we rename this to match the others?

Reply:
Done!